kura/org.eclipse.kura.asset.provider/src/main/java/org/eclipse/kura/asset/provider/AssetConstants.java

```diff
@@ -0,0 +1,107 @@
+/*******************************************************************************
+ * Copyright (c) 2016, 2017 Eurotech and/or its affiliates and others
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ *******************************************************************************/
+package org.eclipse.kura.asset.provider;
+
+import org.eclipse.kura.type.DataType;
+
+/**
```

Comment:
I agree with you on the fact that moving AssetConstants out of the APIs will increase code duplication, but the constants in this class are all related to the particular configuration format used by BaseAsset, they are not related with the Asset itself.
The only bundles that are interested in these constants are the BaseAsset bundle and the web UI bundle. 
The code duplication in the web bundle was there even before this PR because the Asset related constants were hardcoded in the PropertiesUi.java, I think this duplication cannot be avoided if we use GWT.